src/bidiMapper/domains/script/Realm.ts

```diff
@@ -265,47 +238,39 @@ export class Realm {
     return this.#origin;
   }
 
-  get type(): Script.RealmType {
-    return this.#type;
+  get source(): Script.Source {
+    return {
+      realm: this.realmId,
+      ...(this instanceof WindowRealm
+        ? {context: this.browsingContext.id}
+        : {}),
+    };
   }
 
   get cdpClient(): CdpClient {
     return this.#cdpClient;
   }
 
-  get realmInfo(): Script.RealmInfo {
-    switch (this.type) {
-      case 'window':
-        return {
-          realm: this.realmId,
-          origin: this.origin,
-          type: this.type,
-          context: this.browsingContextId,
-          ...(this.sandbox === undefined ? {} : {sandbox: this.sandbox}),
-        };
-      default:
-        return {
-          // TODO: add proper owners.
-          //  https://github.com/GoogleChromeLabs/chromium-bidi/issues/1667
-          owners: [] as any as [Script.Realm],
-          realm: this.realmId,
-          origin: this.origin,
-          type: this.type,
-        };
-    }
+  abstract get associatedBrowsingContexts(): BrowsingContextImpl[];
```

Comment:
Can a realm have more then 1 associated browsing context?

Reply:
Yes, for example a shared worker can be associated with multiple browsing contexts.

Reply:
ack